### PR TITLE
Launcher UI

### PR DESCRIPTION
**Pull request by hakasapl**
**WIP**

- [x] Launcher UI Layout and Functionality
- [x] Move `ParallaxGenConfig` and `ParallaxGenUI` to `ParallaxGen` instead of `ParallaxGenLib`
- [x] Create new parameter schema
- [x] Parameter schema persistence
- [x] Form validation (error messages and validation happen in the UI instead of the terminal)
- [x] Help tooltips
- [ ] Tests for ParallaxGenConfig
- [ ] Tests for ParallaxGenUI (the logic part)
- [x] Doxygen comments

### CODE REVIEW

**Comment by Sebastian-81** [comment]
- [ ] Nexus description needs to be updated

**Comment by hakasapl** [comment]
> * [ ]  Nexus description needs to be updated

That's an interesting question - should we bu updating the description in main according to what is currently in main? Or updating it according to what it should have right now?

**Comment by Sebastian-81** [comment]
> > * [ ]   Nexus description needs to be updated
> 
> That's an interesting question - should we bu updating the description in main according to what is currently in main? Or updating it according to what it should have right now?

Should be based on the version/source code imho. If it needs to be updated for an older release, like 0.6.0, there could be a release branch which is anyway needed for hotfixes

**Comment by hakasapl** [comment]
Closed in favor of https://github.com/hakasapl/ParallaxGen/pull/66

File: ParallaxGenLib/src/ParallaxGenPlugin.cpp [hunk 1]
```diff
@@ -328,9 +323,16 @@ void ParallaxGenPlugin::processShape(const NIFUtil::ShapeShader &AppliedShader,
     PatcherShader::PatcherMatch WinningMatch;
     wstring DecisionMod;
 
-    const auto MeshFilePriority = PGC->getModPriority(PGD->getMod(NIFPath));
+    int MeshFilePriority = -1;
```
[the remark this answers is not in the document]
Comment (hakasapl) [reply ParallaxGenLib/src/ParallaxGenPlugin.cpp:346]: this is address in https://github.com/hakasapl/ParallaxGen/pull/66

File: ParallaxGen/include/ParallaxGenUI.hpp [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+   *
+   * @param Event wxWidgets event object
+   */
+  void onClose(wxCloseEvent &Event);
+};
+
+/**
+ * @brief wxDialog that allows the user to sort the mods in the order they want
+ */
+class ModSortDialog : public wxDialog {
```
[the remark this answers is not in the document]
Comment (hakasapl) [reply ParallaxGen/include/ParallaxGenUI.hpp:168]: will do

**Pull request closed without merging**